sdk/azcore/tracing/tracing.go

```diff
@@ -0,0 +1,168 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// Package tracing contains the definitions needed to support distributed tracing.
+package tracing
+
+import (
+	"context"
+)
+
+// ProviderOptions contains the optional values when creating a Provider.
+type ProviderOptions struct {
+	// for future expansion
+}
+
+// NewProvider creates a new Provider with the specified values.
+//   - impl is the underlying implementation for creating Tracer instances
+//   - options contains optional values; pass nil to accept the default value
+func NewProvider(impl func(name, version string) Tracer, options *ProviderOptions) Provider {
+	return Provider{
+		impl: impl,
+	}
+}
+
+// Provider is the factory that creates Tracer instances.
+// It defaults to a no-op provider.
+type Provider struct {
+	impl func(name, version string) Tracer
```

Comment:
Is there ever a possibility we might return an error when constructing a Tracer? Or is there another way to signal problems?

Reply:
Good question.  This implementation was based on the [OTel for Go](https://pkg.go.dev/go.opentelemetry.io/otel) implementation where these are no-fail.  Our track 1 tracing implementation is based on Open Census and there too these operations are no-fail.

Conceptually I think this makes sense.  Given it's diagnostic tooling, would you want to fail your APIs and potentially bring down the process if your tracing provider is having a bad day?  @lmolkova probably has better insight here.

Reply:
It looks like there's global error handling, at least for OTel.

https://pkg.go.dev/go.opentelemetry.io/otel#ErrorHandler

Reply:
I think it's perfectly okay to say "handle this in your implementation" and stay out of the game.